packages/exhibitor/extra/dcos_zk_backup.py

```diff
@@ -0,0 +1,357 @@
+
+    print('Validate that ZooKeeper is not running')
+    if _is_zookeeper_running(verbose):
+        sys.stderr.write('ZooKeeper must not be running. Aborting.\n')
+        sys.exit(1)
+
+    print('Moving ZooKeeper files temporarily to {tmp_zookeeper_dir}'.format(
+        tmp_zookeeper_dir=tmp_zookeeper_dir,
+    ))
+    # We copy files rather than move them so that if this script is interrupted
+    # while running, no data is lost inplace.
+    # However, this has a downside that we may use a lot of disk space -
+    # we have seen ZooKeeper files up to 20 GB in size.
+    _copy_dir_and_preserve_ownership(
+        src=zookeeper_dir,
+        dst=tmp_zookeeper_dir,
+        verbose=verbose,
+    )
+    shutil.rmtree(path=str(zookeeper_dir))
+
+    print('Restoring {zookeeper_dir} from backup {backup}'.format(
+        zookeeper_dir=zookeeper_dir,
+        backup=backup,
+    ))
+    with tarfile.open(name=str(backup), mode='r:gz') as tar:
+        tar.extractall(path=str(EXHIBITOR_DIR))
+        if verbose:
+            tar.list()
+
+    print('Deleting temporary files in {tmp_zookeeper_dir}'.format(
+        tmp_zookeeper_dir=tmp_zookeeper_dir,
+    ))
+    shutil.rmtree(path=str(tmp_zookeeper_dir), ignore_errors=True)
+
+    print('Local ZooKeeper instance restored successfully')
+
+
+def _non_existing_file_path_existing_parent_dir(value: str) -> Path:
+    """
+    Validate that the value is a path to a file which does not exist
+    but the parent directory tree exists.
+    """
+    path = Path(value)
+    if path.exists():
+        raise ArgumentTypeError('{} already exists'.format(path))
+    if not Path(path.parent).exists():
+        raise ArgumentTypeError(
+            '{} parent directory does not exist'.format(path),
+        )
+    return path.absolute()
+
+
+def _existing_file_path(value: str) -> Path:
+    """
+    Validate that the value is a file which does exist on the file system.
+    """
+    path = Path(value)
+    if not path.exists():
+        raise ArgumentTypeError('{} does not exist'.format(path))
+    if not path.is_file():
+        raise ArgumentTypeError('{} is not a file'.format(path))
+    return path.absolute()
+
+
+def _existing_dir_path(value: str) -> Path:
+    """
+    Validate that the value is a directory which does exist on the file system.
+    """
+    path = Path(value)
+    if not path.exists():
+        raise ArgumentTypeError('{} does not exist'.format(path))
+    if not path.is_dir():
+        raise ArgumentTypeError('{} is not a directory'.format(path))
+    return path.absolute()
+
+
+class DCOSZooKeeperCli:
+    """
+    Minimal CLI to backup/restore the local DC/OS ZooKeeper instance.
+    """
+
+    def __init__(self) -> None:
+        """
+        Present CLI command choices.
+        """
+        parser = ArgumentParser(
+            description=(
+                'Command line utility to more safely interact with the local '
+                'ZooKeeper instance on DC/OS master nodes'
```

Comment:
Also this description should be rewritten to summarize the actual purpose as concisely as possible. It should probably contain the words "backup" and "restore" :-).